### PR TITLE
doc: rewrite GOVERNANCE.md

### PR DESCRIPTION
**Pull request by ghost**
this is an __Extremely Early__ write-up for what our future governance structure could look like. there's currently multiple things missing, and it's probably full of inconsistencies and missing links, but it's a start.

TODO:

- [x] write how teams are elected
- [x] probably write more about efa's
- [x] write about github rights management

### CODE REVIEW

**Comment by ghost** [comment]
this would close #2

File: GOVERNANCE.md
Comment (addaleax) [inline GOVERNANCE.md]: How about 3 or 5 to always have an odd number?

**Comment by Qard** [comment]
The term "member" feels a little more exclusionary than collaborator to me, but that could just be my own perspective. Can't really think of a better word than collaborator. 🤔 

Some statement on what "consensus" is would be helpful here. ie: 50%? 70%? 100%? When would EFA step in?

**Comment by ghost** [comment]
@Qard idk in my view collaborator was always too connected to actual technical stuff, but i'm open to terminology changes

**Comment by Qard** [comment]
That's fair. Not sure what would be a better word.

**Comment by Qard** [comment]
Maybe just defining the openness of the definition of "members" would work?

**Comment by ghost** [comment]
sure, i could add a `Why not collaborator?` segment and explain that there

**Comment by ghost** [comment]
regarding the definition of consensus, i believe the TSC has always required a simply majority vote?

File: GOVERNANCE.md
Comment (lekoder) [inline GOVERNANCE.md]: This words "project governance" as part of "technical issues" - intentional?

File: GOVERNANCE.md
Comment (lekoder) [inline GOVERNANCE.md]: I think you are using "Member" now, link still has "Collaborators"

File: GOVERNANCE.md
Comment (shadowcodex) [inline GOVERNANCE.md]: I think we should point out that changes and issues submitted by sponsors will not have increased priority over that of members and the community as a whole.

**Comment by shadowcodex** [comment]
I think we should point out that changes and issues submitted by sponsors will not have increased priority over that of members and the community as a whole. 

This should be part of the consensus seeking process?

File: GOVERNANCE.md
Comment (lekoder) [inline GOVERNANCE.md]: As per discord discussion, I think "decisive vote" is a better term here; It implies authority only in case of lack of consensus, which seems to be a core of EFA.

File: GOVERNANCE.md
Comment (shadowcodex) [inline GOVERNANCE.md]: I know part of this document is to figure out how members get onto each commit so I wanted to start a review thread within this pull to talk about that.

My thoughts (opinions)(perception):

- Members elect EFA's
- Possibly EFA's elect TC and CC members Alternatively TC & CC are elected by community members.
- Upon term expiring for an EFA then a review should kick off to remove or add members to the TC or CC (Similar to a president coming in and out and re-evaluating the cabinet staff?)
- TC and CC should have terms as well, but may be re-elected. This will force a periodic review of the member if we aren't reviewing them via the EFA elections.

Other thoughts? These are just ideas and should be taken as that at face value.

File: GOVERNANCE.md
Comment (lekoder) [reply GOVERNANCE.md]: This raises interesting issue - in case of different opinions of EFA there is a vote? If not, how EFA for specific issue is choosen?

**Comment by scotttrinh** [comment]
Spit-balling: If the only duty of the EFA is to break a deadlock, what if EFA's were elected ad hoc, as needed? Mutually agreed upon third-parties?

File: GOVERNANCE.md
Comment (TimothyGu) [inline GOVERNANCE.md]: Can I get a clarification on what "adjacent projects" may be? Projects under the ayojs GitHub org? Projects used in Ayo.js? Node.js?

File: GOVERNANCE.md
Comment (TimothyGu) [inline GOVERNANCE.md]: I don't remember this rule actually being invoked in Node.js. If this is the case, we should reconsider continuing to have it.

I'm not with Node.js project long enough though, so if this is not the case *please* correct me.

File: GOVERNANCE.md
Comment (TimothyGu) [inline GOVERNANCE.md]: Is how TC members are added documented anywhere (or will they be)? Will the TC have its own repo like Node.js TSC/CTC does?

File: GOVERNANCE.md
Comment (TimothyGu) [inline GOVERNANCE.md]: Couldn't parse the last sentence, missing a verb in the main clause it seems.

File: GOVERNANCE.md
Comment (TimothyGu) [inline GOVERNANCE.md]: EFAs too?

**Review by TimothyGu — COMMENTED** [review]
I like the simple terminology of TC and CC, though I have to mull on the properties and responsibilities of EFAs a bit...

File: GOVERNANCE.md
Comment (ghost) [reply GOVERNANCE.md]: makes sense, i don't remember anything related to that rule being enforced either

File: GOVERNANCE.md
Comment (ghost) [reply GOVERNANCE.md]: it will be, yeah!

File: GOVERNANCE.md
Comment (ghost) [inline GOVERNANCE.md]: i'm not at all satisfied with the phrase `managing a part of the project`, but i can't think of anything else. thoughts?

Comment (TimothyGu) [reply GOVERNANCE.md]: s/a part/particular facets/?

**Comment by ShaneCurcuru** [comment]
> ...point out that changes and issues submitted by sponsors will not have increased priority over that of members and the community as a whole

That's a key differentiator between where it seems Ayo is going versus the Node Foundation's governance.  Possibly the biggest difference, at least from the outside perspective.

> ... EFA's were elected ad hoc, as needed? Mutually agreed upon third-parties?

That's a great idea for an outside perspective, as long as the bulk of your TC/CC members 1) are confident you can find third parties who can show up and understand the issues to make decisions, and 2) are confident the decisions will be in the spirit of the project.  It's hard for an outside party to really understand the hard issues when there are disagreements - not impossible, but difficult.  Super interesting idea though.

**Comment by scotttrinh** [comment]
@ShaneCurcuru 

I imagine the "third parties" would still be parties involved in the project, and are from, perhaps, different committees/boards/areas of influence within the community. They would be the same pool of people that would be elected by the other means currently being suggested. Maybe even community members, library authors, contributors, etc. For particularly hairy situations, this could go to multiple levels: elect people who elect the EFA's. But, I suppose the thrust of my suggestion is that there is no need for a standing set of EFA's, and they have the potential to have outsized influences if EFA is a position someone might serve on across multiple, consecutive issues.

Certainly is worth looking at the history of mutual, non-state arbitration to find prior art on this to inform our decision. Think juries, non-state courts, courts of cultural diasporic peoples, etc.

**Comment by scotttrinh** [comment]
Trying to avoid a classic case of "who watches the watchers?" and keep from saddling the same group of people with the burden of arbitrating for a whole term.

File: GOVERNANCE.md
Comment (ghost) [reply GOVERNANCE.md]: What about "directing", "leading", or "stewarding" different parts (or facets, like @TimothyGu suggests) of the project? Directing and leading might be too top-down, and stewarding might go too far in the other direction, so maybe "managing" is best...

**Comment by abritinthebay** [comment]
Would this document also cover the concept that was discussed of separate WG's for each major feature? I guess that might come under _Committees_ but I'm not sure if consensus was reached around that idea in the first place.

**Comment by ghost** [comment]
@abritinthebay i was gonna add that but i'm not sure if we've reached consensus on what form we want working groups to exist (if at all) yet?

**Comment by abritinthebay** [comment]
@pup makes sense! Was mostly curious 😄. It's one of the things I'm most excited about for Ayo tbh.

Sounds like it would belong here but in a future iteration.

File: GOVERNANCE.md
Comment (varjmes) [inline GOVERNANCE.md]: can we capitalise the "o" here in "occasionally"

File: GOVERNANCE.md
Comment (varjmes) [inline GOVERNANCE.md]: I'd just take this out for now.

File: GOVERNANCE.md
Comment (varjmes) [inline GOVERNANCE.md]: "An" EFA?

Comment (varjmes) [reply GOVERNANCE.md]: Also I'm interested how an EFA cannot be a member of the Core team. With our current size, this seems hard to do.

**Review by varjmes — CHANGES_REQUESTED** [review]

File: GOVERNANCE.md
Comment (scotttrinh) [reply GOVERNANCE.md]: Yeah, I agree with @varjmes here, I'd say the only restriction for the EFA(s) is that they haven't already been apart of the issue in question, which will be hard enough to find early on without imposing any further restriction.

File: GOVERNANCE.md
Comment (ghost) [reply GOVERNANCE.md]: okay!

**Comment by ghost** [comment]
i'm giving this PR 2 more hours (roughly 6pm CEST) until i land it! cc @ayojs/core

**Review by varjmes — APPROVED** [review]

**Review by sandfox — APPROVED** [review]

**Comment by ghost** [comment]
landed in ebca59b026b592d5c5de480c4aa7d2c701214b6b!!!! :tada:

**Pull request closed without merging**
